File: src/Api/Models/Recipe.cs
using System;
using System.Collections.Generic;

namespace Kokbok.Api.Models;

public sealed class Recipe
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public required User Owner { get; set; }
    public required DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset? DeletedAt { get; set; }
    public List<User> SharedWith { get; } = [];
    public Visibility Visibility { get; set; } = Visibility.Private;
    public Source? Source { get; set; }
    public Asset? Image { get; set; }

    public List<Section<Instruction>> Instructions { get; } = [];
    public List<Section<Ingredient>> Ingredients { get; } = [];
}

File: src/Frontend/src/lib/models/Recipe.ts
import type { Ingredient } from "./Ingredient";
import type { Instruction } from "./Instruction";
import type { Section } from "./Section";
import type { Source } from "./Source";
import type { User } from "./User";

export declare interface LightRecipe {
  id: string;
  name: string;
  description?: string;
  imageId?: string;
  owner: User;
}

export declare interface Recipe extends LightRecipe {
  source?: Source;
  ingredients: Section<Ingredient>[];
  instructions: Section<Instruction>[];
}
